core/types/block.go

```diff
@@ -279,6 +279,7 @@ func CopyHeader(h *Header) *Header {
 		copy(cpy.Extra, h.Extra)
 	}
 	if h.WithdrawalsHash != nil {
+		cpy.WithdrawalsHash = new(common.Hash)
```

Comment:
So, previously, the copied header shares them same withdrawalsHash pointer with the original one?

Reply:
Yes, its not a problem, since its only ever read in rlpHash, but still kinda unsafe I think

Reply:
Sure, we should fix it

Reply:
First this looked wrong to me, so had to dig a bit. 


After the `cpy=*h`, the `cpy` and the `h` have a pointer to the same hash. 
The `cpy` have one pointer, the `h` have another pointer, and both pointers have the same value. At this point, we know it's not the `nil`-value. 

Previously, we deferred the `cpy` pointer-value, and overwrote it with the contents of the `h` pointer value. Which seems to have been a complete no-op, as far as I can tell. 

So yeah, I totally agree.